### PR TITLE
Sign out - Sign in sanity check & nightly sanity check

### PR DESCRIPTION
**Pull request by ouchadam**
- Reintroduces the sign out/re-signin flow to the sanity tests 
- Fixes multiple sign outs occurring whilst signing out during a sync task execution 
- Changes the Sanity check to execute on a nightly basis instead of on each pull request push
  - Updates the action runner to the use hardware accelerated OSX runner 
  - Fixes bad argument placement for the `--no-rate-limit` option

| GIF | 
| --- | 
|after-sanity-2

### CODE REVIEW

File: matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/SessionManager.kt
Comment (bmarty) [inline matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/SessionManager.kt:58]: To avoid usage of `!!`, maybe store `sessionComponents[sessionId]` in a val and test for nullability?

File: vector/src/androidTest/java/im/vector/app/ui/robot/ElementRobot.kt
Comment (bmarty) [inline vector/src/androidTest/java/im/vector/app/ui/robot/ElementRobot.kt:104]: That's a clever way to do it, but which will never make the test fail if it should fail. Maybe add a param `expectSignOutWarning: Boolean` to compare with?

File: .github/workflows/sanity_test.yml
Comment (bmarty) [inline .github/workflows/sanity_test.yml:52]: The issue `--no-rate-limit` has already been fixed somewhere, probably in a not-merged PR... Good catch anyway

**Review by bmarty — APPROVED** [review]
Thanks, a few minor remarks, not blocking the merge

File: matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/SessionManager.kt
Comment (ouchadam) [reply matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/SessionManager.kt:58]: good point :+1:

File: vector/src/androidTest/java/im/vector/app/ui/robot/ElementRobot.kt
Comment (ouchadam) [reply vector/src/androidTest/java/im/vector/app/ui/robot/ElementRobot.kt:104]: I originally had the `hasSentMessages` as a parameter but the verification popup and exit flow isn't always consistent 

Adding a warning log sounds like the best of both worlds, will add :+1:

File: .github/workflows/sanity_test.yml
Comment (ouchadam) [reply .github/workflows/sanity_test.yml:52]: yeahh it was fixed here but it caused 3/4 hour+ runs so I ended up not merging

File: matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/SessionManager.kt
Comment (ouchadam) [reply matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/SessionManager.kt]: done 9c1d6e0484c33d6a471790a9017db57954babb40

**Comment by ouchadam** [comment]
@bmarty once the CI is happen I'll merge and hopefully we'll have some results tomorrow morning :crossed_fingers:

**Pull request merged**
